src/Akka.Persistence.Azure/Journal/AzureTableStorageJournal.cs

```diff
@@ -87,20 +89,10 @@ public AzureTableStorageJournal(Config config = null)
 
             _log.Debug("Entering method ReadHighestSequenceNrAsync");
 
-            var sequenceNumberQuery = GenerateHighestSequenceNumberQuery(persistenceId);
-            TableQuerySegment<HighestSequenceNrEntry> result = null;
-            long seqNo = 0L;
-
-            do
-            {
-                result = await Table.ExecuteQuerySegmentedAsync(sequenceNumberQuery, result?.ContinuationToken);
-
-                if (result.Results.Count > 0)
-                {
-                    seqNo = Math.Max(seqNo, result.Results.Max(x => x.HighestSequenceNr));
-                }
-            } while (result.ContinuationToken != null);
-
+            var seqNo = await HighestSequenceNumberQuery(persistenceId)
+                .Select(entity => entity.GetInt64(HighestSequenceNrEntry.HighestSequenceNrKey).Value)
+                .AggregateAsync(0L, Math.Max);
```

Comment:
To compute the highest sequence number, is that a constant time operation (peeking the top of the partition) or do we have to scan a portion of it? Been a while since I've looked at this code.

Reply:
The query grabs all of the persistence id entries filtered to the last row key for that id, it then compares all of them for the highest sequence numbers from those entries.
In theory, this will be very slow for persistence table with lots of persistence ids